doc/sg/c_cpp_coding_style.md

```diff
@@ -489,6 +489,26 @@ This is especially important in the case of a polyglot header, since function-lo
 Functions marked `static` must not be marked `inline`.
 The compiler is capable of inlining static functions without the `inline` annotation.
 
+### `volatile` Type Qualifier
+
+Do not use `volatile` in production, i.e. non-test, silicon creator code unless you are implementing a library explicitly for this purpose like `sec_mmio`, `abs_mmio`, or `hardened`.
+Specifically, do not use `volatile` for
+
+*   Accessing memory-mapped registers,
+    *   _Instead, use `sec_mmio` or `abs_mmio`_.
+*   Hardening purposes.
+    *   _Instead, use the primitives in `hardened.h`._
+*   Variables initialized outside the lifetime of the program, e.g. a pointer to some data in flash or retention SRAM, or
+    *   _Instead, declare as non-`const `and non-`volatile`. Use a getter to enforce `const`-ness if needed._
+*   Pointers created out of thin air.
+    *   _Compiler cannot optimize the first access (assuming that the pointer doesn't point to something already known by the compiler)._
+
+Do use `volatile` if a variable is shared between an interrupt/exception handler and the rest of the on-target test program to ensure correctness.
+*   _Since silicon creator code does not use handlers in the conventional sense, this essentially limits `volatile` usage to on-target tests._
```

Comment:
I'd maybe just leave it at the first sentence for this page rather than duplicating the list. There's already a link to the other page for more details, and that reduces the risk they will get out of sync.
```suggestion
```

Reply:
Good point, done. Thanks!